### PR TITLE
Fixes #745: Issue created with inherits_containers

### LINKED ISSUE #745: Inherits container bug
Hi all,

On Sonata 3.x there was a feature inherts_containers when defining templates. When implementing [2aa1cb3002e58ffe9d0493be5ca30003ec62a75b](https://github.com/sonata-project/SonataPageBundle/commit/2aa1cb3002e58ffe9d0493be5ca30003ec62a75b) commit the core code for this feature was removed. But the rest wasn't cleaned from documentation? So was this intentional? And if so why do the docs still show this? Or was this just a mistake?

The following code was removed:

```
if (isset($template['inherits_containers'])) {
    $inherits = $template['inherits_containers'];
    if (!isset($templates[$inherits])) {
        throw new InvalidConfigurationException(sprintf('Template "%s" cannot inherits containers from undefined template "%s"', $id, $inherits));
    }
    $template['containers'] = array_merge($templates[$inherits]['containers'], $template['containers']);
}
```

### PR DESCRIPTION
**Pull request by pix-art**
See ticket #745 

https://sonata-project.org/bundles/page/3-x/doc/reference/advanced_configuration.html

Changelog
-----------

```
### Fixed
- Fixed 'inherits_containers' feature
```

Subject
--------

inherits_containers feature was malfunctioning due to the core code being delete in an earlier commit

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -213,6 +213,14 @@ public function getConfigTreeBuilder()
                                 continue;
                             }
 
+                            if (isset($template['inherits_containers'])) {
+                                $inherits = $template['inherits_containers'];
+                                if (!isset($templates[$inherits])) {
+                                    throw new InvalidConfigurationException(sprintf('Template "%s" cannot inherits containers from undefined template "%s"', $id, $inherits));
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php]: Inherit without an s

Comment (greg0ire) [inline DependencyInjection/Configuration.php]: Also, please linebreak this line.

Comment (pix-art) [inline DependencyInjection/Configuration.php]: Docs:
inherits_containers: parent container

I think the reason this had an s in it was so you get **"template X inherists containers from Y"**

Comment (greg0ire) [inline DependencyInjection/Configuration.php]: Trust me, no "s" here. The verb here is "cannot" and it doesn't take an "s" either.

Comment (pix-art) [inline DependencyInjection/Configuration.php]: Tests added :-)

**Comment by pix-art** [comment]
Tests have been added and i updated the PR

**Comment by greg0ire** [comment]
I had a look at the docs, and have no idea what `inherits_container` is about… could you maybe improve the docs to better show what this does and how it is supposed to be used?

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (pix-art) [inline DependencyInjection/Configuration.php]: Oooh my, I finally get which 's' you ment of course that's just bad english. My bad. both fixed

**Comment by pix-art** [comment]
Updated docs and typos

**Pull request closed without merging**